### PR TITLE
Web console: improve how code is imported, use API instance

### LINKED ISSUE #10402: Make use of a single instance of Axios in the web-console
### Motivation
Make use of a single instance of Axios in the web-console.

### Proposed changes
This will lead to changes in all the API calls made using Axios's multiple instances to a single instance.

Any changes made to user-facing interfaces?
- API network calls only

### Rationale
- This can lead to added benefits of the Axios library where anyone can add or remove network-based configurations.
- Can help in the separation of all network requests from the views.
- Removal of multiple instances of Axios which will reduce the import size.

### Operational impact
- Is anything going to be deprecated or removed by this change? How will we phase out old behavior?
No, we will be only phasing out the multiple instances of Axios which will reduce the import size.

- Is there a migration path that cluster operators need to be aware of?
No

- Will there be any effect on the ability to do a rolling upgrade, or to do a rolling _downgrade_ if an operator wants to switch back to a previous version?
No

### Test plan
New unit tests not applicable. However, should pass all the existing unit tests.

### Future work
We can separate all the network requests out of the views and keep them under the respective names.

Sincerely
Ashish Kapoor

### PR DESCRIPTION
**Pull request by vogievetsky**
This is a misc cleanup PR to make the code structure more elegant and clean using fewer globals.
Also fixes #10402

### CODE REVIEW

File: web-console/src/components/show-log/show-log.tsx [hunk 1]
```diff
@@ -17,13 +17,12 @@
...
 import { AnchorButton, Button, ButtonGroup, Intent, Switch } from '@blueprintjs/core';
-import axios from 'axios';
 import copy from 'copy-to-clipboard';
 import React from 'react';
 
 import { Loader } from '../../components';
-import { AppToaster } from '../../singletons/toaster';
-import { UrlBaser } from '../../singletons/url-baser';
+import { Api, AppToaster } from '../../singletons';
+import { UrlBaser } from '../../singletons';
```
Comment (jgoz) [inline web-console/src/components/show-log/show-log.tsx:25]: Minor: these imports can be merged

File: web-console/src/components/show-json/show-json.tsx [hunk 2]
```diff
@@ -17,13 +17,12 @@
...
 import { Button, ButtonGroup, Intent, TextArea } from '@blueprintjs/core';
-import axios from 'axios';
 import copy from 'copy-to-clipboard';
 import React from 'react';
 
 import { useQueryManager } from '../../hooks';
-import { AppToaster } from '../../singletons/toaster';
-import { UrlBaser } from '../../singletons/url-baser';
+import { Api, AppToaster } from '../../singletons';
+import { UrlBaser } from '../../singletons';
```
Comment (jgoz) [inline web-console/src/components/show-json/show-json.tsx:25]: Minor: these imports can be merged

File: web-console/src/entry.ts [hunk 3]
```diff
@@ -69,16 +67,23 @@ if (typeof consoleConfig.title === 'string') {
...
 if (consoleConfig.customHeaderName && consoleConfig.customHeaderValue) {
-  axios.defaults.headers.common[consoleConfig.customHeaderName] = consoleConfig.customHeaderValue;
+  apiConfig.headers.common[consoleConfig.customHeaderName] = consoleConfig.customHeaderValue;
 }
 if (consoleConfig.customHeaders) {
-  Object.assign(axios.defaults.headers, consoleConfig.customHeaders);
+  Object.assign(apiConfig.headers, consoleConfig.customHeaders);
 }
+
+Api.initialize(apiConfig);
```
Comment (jgoz) [inline web-console/src/entry.ts:85]: 👍 nice

File: web-console/src/views/load-data-view/load-data-view.tsx [hunk 4]
```diff
@@ -121,8 +120,8 @@ import {
   upgradeSpec,
 } from '../../druid-models';
 import { getLink } from '../../links';
-import { AppToaster } from '../../singletons/toaster';
-import { UrlBaser } from '../../singletons/url-baser';
+import { Api, AppToaster } from '../../singletons';
+import { UrlBaser } from '../../singletons';
```
Comment (jgoz) [inline web-console/src/views/load-data-view/load-data-view.tsx:124]: Minor: merged

**Review by jgoz — COMMENTED** [review]
Nice changes, just some very minor suggestions

File: web-console/src/views/load-data-view/load-data-view.tsx (hunk 4 above)
Comment (vogievetsky) [reply web-console/src/views/load-data-view/load-data-view.tsx:124]: oh yeah, search replace victim

**Review by AshishKapoor — APPROVED** [review]
Now I know what you meant by the instance method. Could have been easily included on https://github.com/apache/druid/pull/10408. Anyway looks good!

LGTM

**Comment by vogievetsky** [comment]
I just felt so bad that my other PRs caused so many conflicts with yours!

**Review by clintropolis — APPROVED** [review]

**Pull request merged**